tests/statespace_test.h

```diff
@@ -387,14 +351,92 @@ R"(20
 30 y_1_2 19
 )";
 
+template <typename Simulator>
+void TestNormAndInnerProduct() {
```

Comment:
As with TestSamplingSmall, it might help to have a test for the inner product methods that is verifiable by hand.

Reply:
Done.